webapp/client/src/pages/TaxNumberPage.js

```diff
@@ -0,0 +1,331 @@
+
+  switch (currentState(taxNumberFormState, fields)) {
+    case TAX_NUMBER_FORM_STATES.NothingSelected:
+      break;
+    case TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists:
+      shownFields = [stateDropDown];
+      break;
+    case TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists:
+      shownFields = [stateDropDown];
+      break;
+    case TAX_NUMBER_FORM_STATES.TaxOfficeSelection:
+      shownFields = [stateDropDown, bufaNrDropdown];
+      break;
+    case TAX_NUMBER_FORM_STATES.TaxNumberInput:
+      shownFields = [stateDropDown, taxNumberInput];
+      break;
+    case TAX_NUMBER_FORM_STATES.RequestNewTaxNumber:
+      shownFields = [
+        stateDropDown,
+        bufaNrDropdown,
+        requestNewTaxNumberCheckbox,
+      ];
+      break;
+    default:
+  }
+
+  return (
+    <>
+      <StepHeaderButtons />
+      <FormHeader {...stepHeader} />
+      <StepForm {...form}>
+        <FormRowCentered>
+          <FormFieldYesNo
+            fieldName="steuernummer_exists"
+            fieldId="steuernummer_exists"
+            value={fields.steuernummerExists.value}
+            label={{
+              text: t("lotseFlow.taxNumber.taxNumberExists.labelText", {
+                count: numberOfUsers,
+              }),
+            }}
+            details={{
+              title: t("lotseFlow.taxNumber.taxNumberExists.help.title"),
+              text: t("lotseFlow.taxNumber.taxNumberExists.help.text"),
+            }}
+            errors={fields.steuernummerExists.errors}
+            onChangeHandler={changeTaxNumberExists}
+          />
+        </FormRowCentered>
+        {shownFields}
+      </StepForm>
+    </>
+  );
+}
+
+const fieldPropType = PropTypes.exact({
+  value: PropTypes.any,
+  errors: PropTypes.arrayOf(PropTypes.string),
+});
+
+const selectionFieldPropType = PropTypes.exact({
+  selectedValue: PropTypes.any,
+  options: PropTypes.arrayOf(
+    PropTypes.exact({
+      value: PropTypes.string,
+      displayName: PropTypes.string,
+    })
+  ),
+  errors: PropTypes.arrayOf(PropTypes.string),
+});
+
+const checkboxPropType = PropTypes.exact({
+  errors: PropTypes.arrayOf(PropTypes.string),
+  checked: PropTypes.bool,
+});
+
+TaxNumberPage.propTypes = {
+  stepHeader: PropTypes.exact({
+    // TODO: define these here, not in Python
+    title: PropTypes.string,
+    intro: PropTypes.string,
+  }).isRequired,
+  form: PropTypes.exact({
+    action: PropTypes.string, // TODO: does this change? if not, define here, not in Python
+    csrfToken: PropTypes.string,
+    showOverviewButton: PropTypes.bool,
+    nextButtonLabel: PropTypes.string, // TODO: define here, not in Python
+  }).isRequired,
+  fields: PropTypes.exact({
+    steuernummerExists: fieldPropType,
+    steuernummer: fieldPropType,
+    bundesland: selectionFieldPropType,
+    bufaNr: selectionFieldPropType,
+    requestNewTaxNumber: checkboxPropType,
+  }).isRequired,
+  taxOfficeList: PropTypes.arrayOf(PropTypes.any).isRequired,
```

Comment:
As far as I know, there is not because there are no dicts in Javascript. However, you could either use objects or we could set it with `PropTypes.exact` to expect an object with bufaNr and name of the tax office:

```
taxOfficeList: PropTypes.arrayOf(PropTypes.exact({
    name: PropTypes.string,
    bufaNr: PropTypes.string,
  })).isRequired,
```

Reply:
Oh yes, I think that's a great suggestion. 👍